source/frontend/src/features/properties/map/acquisition/common/models.ts

```diff
@@ -44,30 +44,58 @@ export class AcquisitionOwnerFormModel {
   id?: number;
   rowVersion?: number;
   acquisitionFileId?: number;
-  lastNameOrCorp1?: string;
-  lastNameOrCorp2?: string;
-  givenName?: string;
-  incorporationNumber?: string;
+  isOrganization: string = 'false';
+  lastNameAndCorpName: string | '' = '';
+  otherName: string | '' = '';
+  givenName: string | '' = '';
+  incorporationNumber: string | '' = '';
+  registrationNumber: string | '' = '';
+  contactEmailAddress: string | '' = '';
+  contactPhoneNumber: string | '' = '';
   address?: OwnerAddressFormModel;
 
   isEmpty(): boolean {
-    return (
-      this.lastNameOrCorp1?.trim() === '' &&
-      this.lastNameOrCorp2?.trim() === '' &&
-      this.givenName?.trim() === '' &&
-      this.incorporationNumber?.trim() === ''
-    );
+    if (this.isOrganization) {
+      return (
+        this.lastNameAndCorpName?.trim() === '' &&
+        this.otherName?.trim() === '' &&
+        this.incorporationNumber?.trim() === '' &&
+        this.registrationNumber?.trim() === ''
+      );
+    } else {
+      return (
+        this.givenName?.trim() === '' &&
+        this.lastNameAndCorpName?.trim() === '' &&
+        this.otherName?.trim() === ''
+      );
+    }
   }
 
   toApi(): Api_AcquisitionFileOwner {
     return {
       id: this.id,
       rowVersion: this.rowVersion,
       acquisitionFileId: this.acquisitionFileId,
-      lastNameOrCorp1: this.lastNameOrCorp1,
-      lastNameOrCorp2: this.lastNameOrCorp2,
-      givenName: this.givenName,
-      incorporationNumber: this.incorporationNumber,
+      isOrganization: this.isOrganization === 'true' ? true : false,
+      lastNameAndCorpName: this.lastNameAndCorpName,
+      otherName: this.otherName,
+      givenName: this.isOrganization === 'true' ? null : this.givenName.trim(),
+      incorporationNumber:
+        this.isOrganization === 'true'
+          ? this.incorporationNumber.trim() === ''
+            ? null
+            : this.incorporationNumber.trim()
+          : null,
+      registrationNumber:
+        this.isOrganization === 'true'
+          ? this.registrationNumber.trim() === ''
+            ? null
+            : this.registrationNumber.trim()
+          : null,
+      contactEmailAddr:
+        this.contactEmailAddress.trim() === '' ? null : this.contactEmailAddress.trim(),
```

Comment:
we are using this same logic multiple times - perhaps we could refactor to a common util function?